server/mdm/apple/apple_mdm.go

```diff
@@ -153,36 +151,48 @@ func (d *DEPService) createDefaultAutomaticProfile(ctx context.Context) error {
 // setupAsst is nil, the default profile is registered. It assigns the
 // up-to-date dynamic settings such as the server URL and MDM SSO URL if
 // end-user authentication is enabled for that team/no-team.
-func (d *DEPService) RegisterProfileWithAppleDEPServer(ctx context.Context, team *fleet.Team, setupAsst *fleet.MDMAppleSetupAssistant) error {
+//
+// It does that registration for all tokens associated in any way with that
+// team - that is, if DEP hosts are part of that team then the token used to
+// discover those hosts will be used to register the profile, and if a token
+// has that team as default team for a platform, it will also be used to
+// register the profile.
+//
+// On success, it returns the profile uuid and timestamp for the specific token
+// of interest to the caller (identified by its organization name).
+func (d *DEPService) RegisterProfileWithAppleDEPServer(ctx context.Context, team *fleet.Team, setupAsst *fleet.MDMAppleSetupAssistant, abmTokeOrgName string) (string, time.Time, error) {
 	appCfg, err := d.ds.AppConfig(ctx)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "fetching app config")
+		return "", time.Time{}, ctxerr.Wrap(ctx, err, "fetching app config")
 	}
 
 	// must always get the default profile, because the authentication token is
 	// defined on that profile.
 	defaultProf, err := d.ds.GetMDMAppleEnrollmentProfileByType(ctx, fleet.MDMAppleEnrollmentTypeAutomatic)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "fetching default profile")
+		return "", time.Time{}, ctxerr.Wrap(ctx, err, "fetching default profile")
 	}
 
 	enrollURL, err := EnrollURL(defaultProf.Token, appCfg)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "generating enroll URL")
+		return "", time.Time{}, ctxerr.Wrap(ctx, err, "generating enroll URL")
 	}
 
 	var rawJSON json.RawMessage
+	var requestedTokenModTime time.Time
 	if defaultProf.DEPProfile != nil {
 		rawJSON = *defaultProf.DEPProfile
+		requestedTokenModTime = defaultProf.UpdatedAt
```

Comment:
Just curious, where does the token mod time get used?

Reply:
It _might_ get used by callers (Ensure...) or be needed in the cron job? I'm not totally sure yet (it does play a bit of a role with the sync cursor in the cron), we can remove it if it's unused once we're closer to the end.